cpp/oneapi/dal/detail/profiler.cpp

```diff
@@ -14,29 +14,358 @@
 * limitations under the License.
 *******************************************************************************/
 
+#include <sstream>
+#include <iomanip>
+#include <iostream>
+#include <stdexcept>
+#include <chrono>
 #include "oneapi/dal/detail/profiler.hpp"
+#include <daal/include/services/library_version_info.h>
+#include <mutex>
 
 namespace oneapi::dal::detail {
-profiler_task profiler::start_task(const char* task_name) {
-    return profiler_task(task_name);
+
+static volatile int onedal_verbose_val = -1;
+
+static bool queue_exists_global = false;
+
+//TODO: add support of file output
+//static volatile char verbose_file_val[PATH_MAX] = {'\0'};
+// #define ONEDAL_VERBOSE_FILE_ENV "ONEDAL_VERBOSE_OUTPUT_FILE"
+
+#define ONEDAL_VERBOSE_ENV "ONEDAL_VERBOSE"
+
+void print_header() {
+    daal::services::LibraryVersionInfo ver;
+
+    std::cerr << "Major version:          " << ver.majorVersion << std::endl;
+    std::cerr << "Minor version:          " << ver.minorVersion << std::endl;
+    std::cerr << "Update version:         " << ver.updateVersion << std::endl;
+    std::cerr << "Product status:         " << ver.productStatus << std::endl;
+    std::cerr << "Build:                  " << ver.build << std::endl;
+    std::cerr << "Build revision:         " << ver.build_rev << std::endl;
+    std::cerr << "Name:                   " << ver.name << std::endl;
+    std::cerr << "Processor optimization: " << ver.processor << std::endl;
+    std::cerr << std::endl;
 }
 
-void profiler::end_task(const char* task_name) {}
+static void set_verbose_from_env(void) {
```

Comment:
Why not to call those directly from DAAL and not duplicate here?

Reply:
Done